plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/graph/LGraphUtil.java

```diff
@@ -866,15 +872,25 @@ public static LNode createExternalPortDummy(final IPropertyHolder propertyHolder
         case NORTH:
             dummy.setProperty(InternalProperties.IN_LAYER_CONSTRAINT, InLayerConstraint.TOP);
             dummy.getSize().x = portSize.x;
+            if (portBorderOffset < 0) {
+                dummy.getSize().y = -portBorderOffset;
+            }
             dummyPort.setSide(PortSide.SOUTH);
             if (!explicitAnchor) {
                 anchor.y = portSize.y;
             }
+            
+            // See comments in case WEST. This partly fixes #680.
```

Comment:
What do you mean by _partly_? Is it part of the fix or does it only solve the issue partly (i.e. we have to do more)?

Reply:
It solves it partly. The the other part of the issue is solved by lines 875-877 above.